### PR TITLE
make all prow controllers support an optional job config flag

### PR DESCRIPTION
**Pull request by krzyzacy**
Also want to unify all the option interfaces, but maybe in a separate PR.

/area prow
/assign @BenTheElder @cjwagner @fejta @stevekuznetsov

### CODE REVIEW

**Comment by krzyzacy** [comment]
/hold
pending on https://github.com/kubernetes/test-infra/pull/8304

**Comment by stevekuznetsov** [comment]
You've got a test error:

```
Could not load config: stat ../../../config: no such file or directory
```

Otherwise 
/lgtm

**Comment by krzyzacy** [comment]
yeah that dir is being created in #8304

File: prow/cmd/tot/main_test.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func TestFallback(t *testing.T) {
 var c *config.Config
 
 func TestMain(m *testing.M) {
-	conf, err := config.Load("../../config.yaml", "")
+	conf, err := config.Load("../../config.yaml", "../../../config")
```
Comment (cjwagner) [inline prow/cmd/tot/main_test.go]: Should the second parameter be `../../../config/jobs`?
I think the BUILD file may need to be updated to make the `config` directory contents a dependency of this test.

Comment (krzyzacy) [reply prow/cmd/tot/main_test.go]: yeah we've bikesheded the subdir structure for a bit, will update this PR

File: prow/cmd/deck/main.go [hunk 2]
```diff
@@ -77,6 +78,7 @@ func (o *options) Validate() error {
 func gatherOptions() options {
 	o := options{}
 	flag.StringVar(&o.configPath, "config-path", "/etc/config/config.yaml", "Path to config.yaml.")
+	flag.StringVar(&o.jobConfigPath, "job-config-path", "", "Path to prow job configs.")
```
Comment (cjwagner) [inline prow/cmd/deck/main.go]: Should we provide a default for this flag like we do for `config-path`? We aren't consistent about this across binaries (`config-path` doesn't always have a default specified).

Comment (krzyzacy) [reply prow/cmd/deck/main.go]: if we do it now it will break other people's prow deployments :-)

Comment (cjwagner) [reply prow/cmd/deck/main.go]: Ah, good point. I forgot that we are still allowing this to be unspecified.

**Comment by krzyzacy** [comment]
/hold cancel
gonna make a few changes to the deployment files next

File: prow/cmd/tot/main.go [hunk 3]
```diff
@@ -63,6 +64,7 @@ func gatherOptions() options {
 	)
 
 	flag.StringVar(&o.configPath, "config-path", "", "Path to prow config.")
```
Comment (kargakis) [inline prow/cmd/tot/main.go]: Open an issue to remove this eventually for tot since it only cares about the jobs.

Comment (krzyzacy) [reply prow/cmd/tot/main.go]: you mean open an issue to remove tot eventually :-)

**Pull request merged**